### PR TITLE
Read fqdn from /etc/hostname when needed

### PR DESCRIPTION
**Pull request by mchf**
bnc#941881

### CODE REVIEW

File: library/types/src/modules/Hostname.rb [hunk 1]
```diff
@@ -159,7 +160,7 @@ def CurrentFQ
         SCR.Execute(path(".target.bash_output"), "hostname --fqdn")
       )
       if hostname_data.nil? || Ops.get_integer(hostname_data, "exit", -1) != 0
-        fqhostname = if SCR.Read(path(".target.stat"), "/etc/hostname").empty?
+        fqhostname = if FileUtils.Exists("/etc/hostname")
```
Comment (ancorgs) [inline library/types/src/modules/Hostname.rb]: Isn't this implementable using plain ruby? We want to get rid of Yast::Fileutils at some point in future.

**Comment by jreidinger** [comment]
LGTM

File: library/types/src/modules/Hostname.rb (hunk 1 above)
Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: I think it is problem during upgrade, where you want to read hostname from target machine

File: library/types/test/hostname_test.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    allow(Yast::SCR)
+      .to receive(:Read)
+      .with(path(".target.string"), "/etc/hostname")
+      .and_return(TEST_HOSTNAME)
+    allow(Yast::FileUtils)
+      .to receive(:Exists)
+      .with("/etc/hostname")
+      .and_return(true)
+
+    expect(Yast::Hostname.CurrentFQ).to eq TEST_HOSTNAME
```
Comment (ancorgs) [inline library/types/test/hostname_test.rb]: For the shake of completeness, what about adding also a test for a successfully `hostname --fqdn` (that will prove, OTOH that /etc/hostname is not directly read in that case).

File: library/types/test/hostname_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    allow(Yast::SCR)
+      .to receive(:Read)
+      .with(path(".target.string"), "/etc/hostname")
+      .and_return(test_hostname)
+    allow(Yast::FileUtils)
+      .to receive(:Exists)
+      .with("/etc/hostname")
+      .and_return(true)
+
+    expect(Yast::Hostname.CurrentFQ).to eq test_hostname
```
Comment (ancorgs) [inline library/types/test/hostname_test.rb]: Better, thanks.

Although being NP, I would prefer to have two different names for each case, to make more clear they are coming from different places. Something like thi
```ruby
let(:command_hostname) { "host.suse.cz" }
let(:etc_hostname) { "host.suse.de" }
```
And then
```ruby
expect(...).to eq command_hostname
...
expect(...).to eq etc_hostname
```

As said. Just NP. I can leave with the current implementation.

**Comment by ancorgs** [comment]
LGTM

**Comment by mchf** [comment]
Thanks for reviews.

**Pull request merged**